pkg/clients/ilmt/ilmt.go

```diff
@@ -67,212 +67,175 @@ func NewClient(config *IlmtConfig) Client {
 	return cli
 }
 
-func (ilmtC *ilmtClient) FetchUsageData(ctx context.Context, dateRange DateRange) (int, string, error) {
-	startDate, _ := time.Parse(REQUIRED_FORMAT, dateRange.StartDate)
-	endDate, _ := time.Parse(REQUIRED_FORMAT, dateRange.EndDate)
-	daysDifference := endDate.Sub(startDate).Hours() / 24
-	fileCounter := 0
-	productUsageTransformedEventJsonAll := make([][]byte, int(daysDifference+1))
-	productUsageTransformedEventJsonAllStr := EMPTY
-	for selectedDate := startDate; !selectedDate.After(endDate); selectedDate = selectedDate.AddDate(0, 0, 1) {
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
 
-		urlForStandaloneProductUsage, err := ilmtC.req.FetchUsageData(ctx, ilmtC.IlmtConfig.Host, ilmtC.IlmtConfig.Token, CRITERIA_STANDALONE, strings.Split(selectedDate.String(), BLANK_VALUE)[0], strings.Split(selectedDate.String(), BLANK_VALUE)[0])
-		if err != nil {
-			log.Fatal(err.Error())
-			return -1, EMPTY, err
-		}
+func (ilmtC *ilmtClient) FetchUsageData(ctx context.Context, dateRange DateRange, selectedDate time.Time) []byte {
```

Comment:
Maybe we can further modularize this `FetchUsageData` code by moving the processing logic (2 big FOR loops) into a separate util function?